### PR TITLE
test: add tests for pruning store

### PR DESCRIPTION
**Pull request by b00f**
## Description

This PR adds tests for pruning store and also refactor config tests.

### CODE REVIEW

File: store/store.go [hunk 1]
```diff
@@ -441,15 +439,11 @@ func (s *store) Prune(resultFunc func(pruned, skipped, pruningHeight uint32)) er
...
-			resultFunc(1, 0, i)
-
-			continue
-		}
-
-		resultFunc(0, 1, i)
+		resultFunc(deleted, i)
 	}
 
+	s.isPruned = true
```
Comment (Ja7ad) [inline store/store.go:445]: This value stores in memory after prune needs to restart node.

**Review by Ja7ad — CHANGES_REQUESTED** [review]

File: store/store.go (hunk 1 above)
Comment (b00f) [reply store/store.go]: Done.

File: store/store.go [hunk 2]
```diff
@@ -415,39 +419,37 @@ func (s *store) IsPruned() bool {
...
-		if err := s.WriteBatch(); err != nil {
+		if err := s.writeBatch(); err != nil {
 			return err
 		}
 
-		if deleted {
-			resultFunc(1, 0, i)
-
-			continue
+		if resultFunc(deleted, i) {
```
Comment (Ja7ad) [inline store/store.go:449]: Does not required return bool and check result func.

I think remove return bool.

File: store/store_test.go [hunk 3]
```diff
@@ -23,8 +23,8 @@ type testData struct {
 func testConfig() *Config {
 	return &Config{
 		Path:               util.TempDirPath(),
-		TxCacheSize:        1024,
-		SortitionCacheSize: 1024,
+		TxCacheWindow:      1024,
```
Comment (Ja7ad) [inline store/store_test.go:26]: Old name is explicit for here, I think size better as window.

**Review by Ja7ad — CHANGES_REQUESTED** [review]

File: store/store_test.go (hunk 3 above)
Comment (b00f) [reply store/store_test.go:26]: This cache keeps transactions if they are not older than 8,640 blocks. We don't have a fixed size but a fixed window.
The old name was not accurate.

File: store/store.go (hunk 2 above)
Comment (b00f) [reply store/store.go:449]: `resultFunc` is not a very good name here; it checks if the loop should continue or not. This prevents a random crash if the user presses Ctrl+C while Prune is working.

**Review by Ja7ad — APPROVED** [review]

**Pull request merged**